.github/workflows/reusable.yml

```diff
@@ -0,0 +1,104 @@
+name: Deploy AKS-Construction
+
+on:
+  workflow_call:
+    inputs:
+      templateVersion:
+        description: 'Template Version'
+        required: false
+        default: '0.8.5-preview'
+        type: string
+      rg:
+        description: 'ResourceGroup name'
+        required: true
+        type: string
+      resourceName:
+        description: 'Unique name all your resources are based on'
+        required: true
+        type: string
+      templateParamFile:
+        description: 'Template Parameters file'
+        required: false
+        type: string
+      templateParams:
+        description: 'Template Paramterss, space seperated key=value'
+        required: true
+        type: string
+      postScriptParams:
+        description: 'Template Paramterss, comma seperated key=value'
+        required: false
+        type: string
+      postScriptInvokeCommand:
+        description: 'Private Cluster, so use invoke commant for postscript'
+        required: false
+        default: false
+        type: boolean
+    secrets:
+      AZURE_CLIENT_ID:
+        required: true
+      AZURE_TENANT_ID:
+        required: true
+      AZURE_SUBSCRIPTION_ID:
+        required: true
+      USER_OBJECT_ID:
+        required: false
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  aksc_reusable_workflow:
+    runs-on: ubuntu-latest
+    steps:
+      # Log into Azure
+    - uses: azure/login@v1
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      # Deploy Bicep file
+    - name: Deploy Bicep
+      env:
+        templateParams: ${{ inputs.templateParams}}
+      run: |
+        if [ "${{ secrets.USER_OBJECT_ID }}" ]; then
+          echo "Substitute _USER_OBJECT_ID_ placeholder value with real guid from secret"
+          templateParams=${templateParams//=_USER_OBJECT_ID_/=${{ secrets.USER_OBJECT_ID }}}
+        fi
+        az deployment group create -g ${{ inputs.rg }} --template-uri https://github.com/Azure/AKS-Construction/releases/download/${{ inputs.templateVersion }}/main.json --parameters ${{ inputs.templateParamFile}} automatedDeployment=true $templateParams
+
+
+    - name: Kubelogin
+      if: inputs.postScriptParams && inputs.postScriptInvokeCommand == false
+      env:
+        kubeloginversion: 'v0.0.12'
+      run: |
+
+        az aks get-credentials -n aks-${{ inputs.resourceName }} -g ${{ inputs.rg }} --overwrite-existing
```

Comment:
It's going to be better to use the AKSName as outputted from the deployment, rather than use convention.
(Also in my pending PR ;))